deeplabcut/live/dlc_live.py

```diff
@@ -178,44 +234,79 @@ def setup_prediction(self, config):
             self.sess, self.inputs, self.outputs = predict.setup_pose_prediction(self.dlc_cfg)
 
     def get_pose(self, frame=None):
+        """
+
+        Parameters
+        ----------
+        frame :class:`numpy.ndarray`
+            image to extract pose. Should be an RGB, uint8 image, but if grayscale (len(frame.shape)== 2)
+            or if given as float will be automatically converted.
+
+        Returns
+        -------
+
+        """
 
         if frame is None:
-            if self.camera is None:
+            if self.frame_method is None:
                 raise DLCLiveException("No frame provided for live pose estimation")
             else:
-                frame = self.camera.frame
+                frame = self.frame_method()
 
-        frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
-        frame = img_as_ubyte(frame) if not self.cfg['cropping'] else img_as_ubyte(frame[self.cfg['y1']:self.cfg['y2'],self.cfg['x1']:self.cfg['x2']])
+        # These conversions can be pretty costly. Usually I like EAFP but afaik tensorflow doesn't handle
+        # it nicely, so let's
+        #   a) allow frame conversion to be handled by whatever is giving them to us
+        #   b) only do unambiguous operations - eg. grayscale to RGB, but how do we know if a (n,m,3) image is RGB or BGR?
+        #   c)  check if we need to 8-bit-ify it before doing so
+        # -jls 2020-03-09
+
+        if self.cfg['cropping']:
+            frame = frame[self.cfg['y1']:self.cfg['y2'],
+                          self.cfg['x1']:self.cfg['x2']]
+
+        # if we don't have a third axis, this is a grayscale image!!!
+        if len(frame.shape) == 2:
```

Comment:
the network requires 3 axis; it assumes RBG image...

Reply:
please revert

Reply:
yes, this checks if it is 2 axis (cheap as free) and if it is the `cv2.COLOR_GRAY2RGB` casts it as 3 axis RGB by du(tri?)plication (relatively cheap). added it b/c the operation is unambiguous (2d arrays are always grayscale) and makes image preprocessing a bit more forgiving, but don't have strong feelings.